Make the tests pass with Sphinx 8.1

mitya57 opened the pull request:
See sphinx-doc/sphinx#12776 which changed the default theme from alabaster to basic and broke the tests.

Fixes #249.

humitos commented on tests/test_urls.py:
```diff
@@ -52,7 +52,7 @@ def test_404_page_not_created(app, status, warning):
     assert not path.exists()
 
-@pytest.mark.sphinx(srcdir=srcdir)
+@pytest.mark.sphinx(srcdir=srcdir, confoverrides={'html_theme': 'alabaster'})
```
Can you put a comment here explaining why we are defining the theme explicitly here and link to the Sphinx PR that performs the change?

humitos reviewed (changes requested):
Looks good to me. The test that builds the documentation is still failing:

```
/home/circleci/project/docs/autoapi/notfound/extension/index.rst:127:Unknown interpreted text role "event". [docutils]
```

Can you take a look at that one?

mitya57 commented:
Fixed the problem with docs.

It happened because docstrings for `clear_doc` and `process_doc` members of OrphanMetadataCollector were inherited from parent class in Sphinx (EnvironmentCollector), and docstrings in Sphinx [use](https://github.com/sphinx-doc/sphinx/blob/v8.1.3/sphinx/environment/collectors/__init__.py#L50) [`:event:`](https://github.com/sphinx-doc/sphinx/blob/v8.1.3/sphinx/environment/collectors/__init__.py#L73) role, which is [private to Sphinx](https://github.com/sphinx-doc/sphinx/blob/v8.1.3/doc/conf.py#L351-L357). So I added explicit docstrings which do not use `:event:`.

Other options would be copying the `:event:` role to our conf.py, or removing those members using `autoapi-skip-member` handler, but I think my solution is the simplest one.

Also added the comment as you suggested.

humitos reviewed (approved):
Excellent! Thanks
